### PR TITLE
Fix markForCheck calls in side bar

### PR DESCRIPTION
**Pull request by ebrehault**

### CODE REVIEW

File: projects/pastanaga/src/lib/sidebar/sidebar.component.ts [hunk 1]
```diff
@@ -179,14 +179,20 @@ export class SidebarComponent implements OnInit, OnDestroy {
...
     private hideBackdrop() {
         if (!!this.backdrop && !!this.backdrop.parentNode) {
             this.backdrop.parentNode.removeChild(this.backdrop);
             this.backdrop = null;
-            this.cdr.markForCheck();
         }
     }
+
+    private markForCheck() {
+        window.setTimeout(() => {
```
Comment (LuisPalomo) [inline projects/pastanaga/src/lib/sidebar/sidebar.component.ts]: Why do we need to do this `setTimeout`? It's like we are doing something wrong :thinking:

Comment (mathilde-pellerin) [reply projects/pastanaga/src/lib/sidebar/sidebar.component.ts]: agree with Luis, it feels unsafe to encapsulate a `markForCheck` in a timeout. so timeout is not working outside zone, so I guess it's ok to keep `markForCheck` and don't use `detectChanges`, but still, why would you have it always in a timeout like that?

File: projects/pastanaga/src/lib/sidebar/sidebar.component.ts [hunk 2]
```diff
@@ -8,7 +8,8 @@ import {
     OnDestroy,
     OnInit, Output,
     Renderer2,
-    ViewEncapsulation
+    ViewEncapsulation,
+    ViewRef
```
Comment (mathilde-pellerin) [inline projects/pastanaga/src/lib/sidebar/sidebar.component.ts]: seems like you're not using this `ViewRef` import

File: projects/pastanaga/src/lib/sidebar/sidebar.component.ts (hunk 1 above)
Comment (ebrehault) [reply projects/pastanaga/src/lib/sidebar/sidebar.component.ts]: I feel the same, but if we don't do that, and if the parent component is doing a `markForCheck` on its side, then we get some Angular view check errors a way or another.

I spent the afternoon on that and could not come up with a better solution.

So I was planning to merge it (and if you have some time, you can try to find a better implementation).

**Pull request merged**